### DISCUSSION TITLE
Why the output of api.Println() with float type is Garbled code？

### DISCUSSION THREAD

**Discussion by George551556** [Consensys/gnark#1241] [General]
```go
api.Println("x", api.Div(1, 54))
```
Just as the code above, it output
> x 6890743126319771829225720327210623638987448051982825626719805021699791563435
Why is it designed this way? I do know that finite fields do not support floating-point calculations, but through experimentation, I found that the dynamic variable values can be floating-point numbers and can participate in other operations normally. However, why is the output of api.Println() is garbled?

**Comment by George551556** [comment]
Please help me!!!

**Reply by ivokub** [reply]
I'm not understanding the question. All operations in a circuit are done in the finite field. Do you want to perform floating-point operations in the circuit?

There is an open issue to implement floating-point operations through a wrapper (https://github.com/Consensys/gnark/issues/209) similarly how we do non-native field operations, but we don't have in currently on the roadmap.

What do you want to prove with gnark, maybe there is a way without needing floating point operations?
